### ISSUE TITLE
Report on using Transkribus for 1957 minutes

### ISSUE THREAD

**Issue by willf** [sacred-harp-minutes/sacred-harp-minutes#21]
Using the Mac version of Transkribus (Transkribus-1.17.0), I uploaded the 1957 TIFF images to the Transkribus server and had it analyze the pages. It took a while (several hours) to do so; it takes two steps, one to upload the data, and another to start the analysis, using the latest Transkribus Typewriter model.

I then spent some time reviewing the analysis it created; I tried correcting a few pages. 

Observations:

1. I saw very few OCR errors. The only minor issue is that it didn't know how to handle the dingbat used to separate minutes (a black circle). 👍🏼 
2.  It creates each line as a separate block or "shape".  This may be just the right thing, but I don't really think so. See Image 1 below.
3. It often splits the ends of lines into two parts, with the _right_ part ordered previous to the _left_ part. These are sometimes hard to spot. This requires two actions to repair: (a) reorder the shapes, (b) merge the shapes. This is certainly doable, but tedious. I think the software is interpreting the right shape as (part of?) a margin note; if this happens, there are several lines in a row in which it happens, and there is a bit of a gutter between the left and right sections. See Image 2. 👎🏼 

Image 1: Each line is a separate block.
<img width="842" alt="CleanShot 2021-11-10 at 13 23 42@2x" src="https://user-images.githubusercontent.com/37049/141171320-c10491c4-f587-4b64-aecc-6841b2b8da81.png">

Image 2: Wrongly ordered shapes, with space gutter.
<img width="500" alt="CleanShot 2021-11-10 at 13 23 07@2x" src="https://user-images.githubusercontent.com/37049/141171218-3b654ab7-28d7-47ab-bb25-dcfb8901b00f.png">

 Unless (2) is very important to @jpkarlsberg, I think it's worth while exploring other OCR options.

**Comment by willf** [comment]
As a quick test of another OCR system, I used [OCROwl](https://owlocr.com) on the '57 minutes, and found (on visual inspection) the same quality OCR.

**Comment by jpkarlsberg** [comment]
This does seem like a potential dealbreaker with the built-in OCR in Transkribus. I'm curious: are you running Tranksribus's layout analysis as a separate step prior to using the Typewriter model? Is it possible that there are multiple layout analysis models in Transkribus that we could play around with? Something to explore?

The polygons Tranksribus creates for lines (2) are neat, but they're not a part of the _output_ we need for this project.

What Transkribus offers that I find compelling is the Lite interface for correcting OCR errors. The polygons/lines are actually an important part of this, because they make possible the UI that enables the person correcting errors to directly compare a line of OCR'd text with the corresponding page region.

I suspect that we have plenty of options for comparable initial OCR quality. What I don't know is whether there's any way for us to take advantage of the OCR correction interface Transkribus offers without using its OCR and layout analysis models. That could be another avenue to explore.

**Comment by willf** [comment]
I'm exploring the other layout detection options. The "Printed Block Detection" might be better.

**Comment by willf** [comment]
Ok, I ran Transkribus (on my own account) on the 1957 minutes with the following changes:

1. I used “Printed Block Detection" for layout analysis
2. I used "Transcribus OCR (Block-segmentation) + HTR + print” for text recognition

This seemed to solve the ‘merge’ problems I was having. It seems to recognized larger blocks of text larger than line size.

See the image below compared to Image 2 above.

Image 3: Correct block segmentation using “Printed Block Detection"
<img width="867" alt="CleanShot 2021-11-12 at 14 50 22@2x" src="https://user-images.githubusercontent.com/37049/141526425-d8ea4948-732e-4e62-a01b-1d4d0b34a079.png">
